### DISCUSSION TITLE
Anyone know how to rename a node's file extension?

### DISCUSSION THREAD

**Discussion by wahidshafique** [gatsbyjs/gatsby#33916] [Help]
I am using [gatsby-source-airtable](https://www.gatsbyjs.com/plugins/gatsby-source-airtable/) and want to rename a file extension that I am pulling in. This is because the node looks like this

(-- added by me, to remove links to direct urls from airtable)
```
{
  id: 'b41ea2ab-0fc1-5020-b012-01db9b656e09',
  children: [],
  parent: null,
  internal: {
    contentDigest: '44c679da489128493bcf51dc44f8ccb4',
    type: 'File',
    mediaType: 'audio/mpeg',
    description: 'File "--/core-fx-wrong.mp3"',
    owner: 'gatsby-source-filesystem',
    counter: 136
  },
  sourceInstanceName: '__PROGRAMMATIC__',
  relativePath: '.cache/caches/gatsby-source-airtable/5c6a15760c352dfb4d4ce47dbd1580a6/core-fx-wrong.mpga',
  extension: 'mpga',
  prettySize: '77.1 kB',
  modifiedTime: '2021-11-09T16:09:16.626Z',
  accessTime: '2021-11-09T16:09:16.285Z',
  changeTime: '2021-11-09T16:09:16.808Z',
  birthTime: '2021-11-09T16:09:16.285Z',
  root: '/',
  dir: '--/workspace-core-frontend/.cache/caches/gatsby-source-airtable/5c6a15760c352dfb4d4ce47dbd1580a6',
  base: 'core-fx-wrong.mpga',
  ext: '.mpga',
  name: 'core-fx-wrong',
  absolutePath: '/--/workspace-core-frontend/.cache/caches/gatsby-source-airtable/5c6a15760c352dfb4d4ce47dbd1580a6/core-fx-wrong.mpga',
  relativeDirectory: '.cache/caches/gatsby-source-airtable/5c6a15760c352dfb4d4ce47dbd1580a6',
  dev: 16777221,
  mode: 33188,
  nlink: 1,
  uid: 502,
  rdev: 0,
  blksize: 4096,
  ino: 15103740,
  size: 77077,
  blocks: 152,
  atimeMs: 1636474156285.4443,
  mtimeMs: 1636474156625.825,
  ctimeMs: 1636474156808.3457,
  birthtimeMs: 1636474156285.4443,
  atime: '2021-11-09T16:09:16.285Z',
  mtime: '2021-11-09T16:09:16.626Z',
  ctime: '2021-11-09T16:09:16.808Z',
  birthtime: '2021-11-09T16:09:16.285Z',
  url: '--/core-fx-wrong.mp3'
}
```

As you can see, the remote url has the extension of `mp3` yet what I see in my `cache` turns out to be an `mpga`. What I want to know is, how can I rename the file extension as its digested. Is there a way to intercept, I tried `onCreateNode` but was unsure on how to proceed in best way.

**Comment by LekoArts** [comment]
You could try going into the source code of the plugin (in node_modules) and add logging to:

https://github.com/jbolda/gatsby-source-airtable/blob/e85ea3c7c9f5c735fe0095dfe2c6e55321ab54d1/gatsby-node.js#L307

To see if at this step the extension gets changed

**Reply by wahidshafique** [reply]
@LekoArts So I logged, and it seems to be changed at that point, converting to an `mpga`. I'm not quite familiar on how to proceed beyond this finding, any further help would be appreciated.
